python-package/lightgbm/basic.py

```diff
@@ -472,7 +485,7 @@ def __pred_for_np2d(self, mat, num_iteration, predict_type):
         else:
             """change non-float data to float data, need to copy"""
             data = np.array(mat.reshape(mat.size), dtype=np.float32)
-        ptr_data, type_ptr_data = c_float_array(data)
+        ptr_data, type_ptr_data, new_data = c_float_array(data)
```

Comment:
@guolinke If we don't need `new_data` maybe it's better to name it like non-used variable:
```
ptr_data, type_ptr_data, _ = c_float_array(data) 
```
?

Reply:
yeah, `new_data` is used to avoid the temporary copy is freed.